### PR TITLE
workflows: Add build events with workflow metadata and per-command status

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch

### CODE REVIEW

File: proto/build_event_stream.proto [hunk 1]
```diff
@@ -213,6 +226,11 @@ message BuildEventId {
     WorkspaceConfigId workspace = 23;
     BuildMetadataId build_metadata = 24;
     ConvenienceSymlinksIdentifiedId convenience_symlinks_identified = 25;
+
+    // BUILDBUDDY EVENTS BELOW
```
[the remark this answers is not in the document]
Comment (bduffany) [reply proto/build_event_stream.proto:230]: Done

File: proto/build_event_stream.proto [hunk 2]
```diff
@@ -750,6 +768,54 @@ message ConvenienceSymlink {
...
+
+// Event describing a workflow command that completed.
+// Note: the event ID holds the invocation ID that identifies the command.
+message WorkflowCommandCompleted {
+  // The overall status of the command. A command was successful if and only if
+  // this value is equal to 0.
+  int32 exit_code = 1;
+
+  // Time in milliseconds since the epoch.
+  int64 finish_time_millis = 2;
```
[the remark this answers is not in the document]
Comment (bduffany) [reply proto/build_event_stream.proto]: Discussed offline - updated `WorkflowCommandCompleted` to be consistent with `TestResult` -- declaring start time and duration

**Review by tylerwilliams — APPROVED** [review]
LGTM (nit about invocation comments on 800 and 200)

File: proto/build_event_stream.proto [hunk 3]
```diff
@@ -750,6 +770,58 @@ message ConvenienceSymlink {
...
+  // Webhook event name that triggered the action. Can be one of "push" or
+  // "pull_request".
+  string action_trigger_event = 4;
+
+  // Metadata for a Bazel invocation created by this workflow action.
+  message InvocationMetadata {
+    // Bazel invocation ID (UUIDV4). Should not be set for bazel commands that
+    // do not create invocations, such as `bazel version`.
+    // Example: "86083af6-4699-4957-a0a3-d6a806104bb3"
+    string invocation_id = 1;
```
[the remark this answers is not in the document]
Comment (bduffany) [reply proto/build_event_stream.proto:809]: Done

**Pull request merged**